### DISCUSSION TITLE
Dnsdist  Per Domain Metrics

### DISCUSSION THREAD

**Discussion by ahmedhkhalil** [PowerDNS/pdns#14868] [Q&A]
Dears,
i have dnsdist and i configure promethous to scrape endpoint for dnsdist metrics , what i need is to have some metric that give me information about Query per second per Domain  , how can i achieve that

**Comment by rgacogne** [comment]
DNSdist doesn't know anything about domains, in the sense that it doesn't know where the zone cuts are (if it sees `a.sub.domain.tld` it does not know if it belongs to the `sub.domain.tld`, `domain.tld` or even `tld` zone) so it's hard to get per domain metrics. We have a QueryCount feature that might help, but it depends on what you trying to achieve: https://dnsdist.org/guides/carbon.html#query-counters

**Reply by ahmedhkhalil** [reply]
does it work for Prometheus or only graphite

**Reply by rgacogne** [reply]
It looks like it only supports Carbon at the moment, although it shouldn't be hard to make it work for Prometheus as well.
